modules/json_form_widget/src/Element/DateRange.php

```diff
@@ -0,0 +1,208 @@
+        $input = [
+          'start' => $start,
+          'end' => $end,
+          'date_range' => $start . '/' . $end,
+        ];
+      }
+    }
+    return $input;
+  }
+
+  /**
+   * Get time formatted as H:i:S.
+   *
+   * @param mixed $time
+   *   Time to format.
+   *
+   * @return mixed
+   *   Formatted time.
+   */
+  public static function getFormattedTime($time) {
+    $formatted = !empty($time) ? $time : '00:00:00';
+    if (strlen($formatted) == 5) {
+      $formatted = $formatted . ':00';
+    }
+    return $formatted;
+  }
+
+  /**
+   * Callback for processing the date_range element.
+   */
+  public static function processDateRange(&$element, FormStateInterface $form_state, &$complete_form) {
+    $element['start_date'] = [
+      '#type' => 'flexible_datetime',
+      '#title' => 'Start date',
+    ];
+    $element['end_date'] = [
+      '#type' => 'flexible_datetime',
+      '#title' => 'End date',
+    ];
+
+    // Add default value for start and end dates.
+    $default = !empty($element['#default_value']) ? $element['#default_value'] : '';
+    $matches = [];
+    if (preg_match('/(.*)\/(.*)/', $default, $matches)) {
+      $start_date = new DrupalDateTime($matches[1], date_default_timezone_get());
+      $element['start_date']['#default_value'] = $start_date;
+      $end_date = new DrupalDateTime($matches[2], date_default_timezone_get());
+      $element['end_date']['#default_value'] = $end_date;
+    }
+    return $element;
+  }
+
+  /**
+   * Validation callback for a date_range element.
+   *
+   * @param array $element
+   *   The form element whose value is being validated.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The current state of the form.
+   * @param array $complete_form
+   *   The complete form structure.
+   */
+  public static function validateRange(array &$element, FormStateInterface $form_state, array &$complete_form) {
+    $input_exists = FALSE;
+    $input = NestedArray::getValue($form_state->getValues(), $element['#parents'], $input_exists);
+    if ($input_exists) {
+      if (isset($input['start_date']) && isset($input['end_date'])) {
+        static::validateInterval($input['start_date'], $input['end_date'], $element, $form_state);
+        return;
+      }
+      static::missingEndDate($input, $element, $form_state);
+      static::missingStartDate($input, $element, $form_state);
+    }
+  }
+
+  /**
+   * Set errors when end date is missing.
+   *
+   * @param array $input
+   *   Array with keys for start_date and end_date.
+   * @param array $element
+   *   The element being validated.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The current state of the form.
+   */
+  public static function missingEndDate(array $input, array $element, FormStateInterface $form_state) {
+    if (!empty($input['start_date']) && empty($input['end_date'])) {
+      $form_state->setError($element['end_date'], t('Please enter an end date.'));
```

Comment:
don't you have `$this->t()` available?

Reply:
No, because it is a static function so we need to use t.